### PR TITLE
[FEATURE] MS-56: implement login / signUp

### PR DESCRIPTION
**Pull request by EugenioPonzone**

### CODE REVIEW

File: controllers/auth.js [hunk 1]
```diff
@@ -1,8 +1,13 @@
+/* eslint-disable max-len */
+/* eslint-disable no-unused-vars */
+// const jwt = require('jsonwebtoken');
```
Comment (BaltazarCortina) [inline controllers/auth.js:3]: Can we delete this line?

File: controllers/auth.js [hunk 2]
```diff
@@ -1,8 +1,13 @@
...
 const Users = require('../models/Users');
 const Firebase = require('../helpers/firebase');
+const Candidates = require('../models/Candidates');
+const Psychologists = require('../models/Administrators');
+const Admins = require('../models/Psychologists');
 
 // USING ASYNC AWAIT
```
Comment (BaltazarCortina) [inline controllers/auth.js:10]: This one too

File: controllers/auth.js [hunk 3]
```diff
@@ -14,20 +19,58 @@ const register = async (req, res) => {
...
+        number: req.body.address.number,
+      },
+      city: req.body.city,
+      province: req.body.province,
+      country: req.body.country,
+      postalCode: req.body.postalCode,
+      birthday: req.body.birthday,
+      pictureUrl: req.body.pictureUrl,
     });
+    const ouser = await newCandidate.save();
```
Comment (BaltazarCortina) [inline controllers/auth.js:41]: Can we change this to user?

File: controllers/auth.js [hunk 4]
```diff
@@ -14,20 +19,58 @@ const register = async (req, res) => {
...
     // Save the new user on DB
     const userSaved = await userCreated.save();
     // Response with the new user created
     return res.status(201).json({
       message: 'User created',
-      data: userSaved,
+      data: ouser,
```
Comment (BaltazarCortina) [inline controllers/auth.js:47]: Also here

File: controllers/auth.js [hunk 5]
```diff
@@ -14,20 +19,58 @@ const register = async (req, res) => {
...
     });
   } catch (error) {
     // Return error
```
Comment (BaltazarCortina) [inline controllers/auth.js:50]: We could get rid of the comments that are not necessary

File: routes/auth.js [hunk 6]
```diff
@@ -1,11 +1,22 @@
...
-const { register } = controller;
+const { register, loginServer } = controller;
 
-router.post('/register', validations.required, register);
+router.post('/register', validations.required, isNotEmpty, validateLength, register);
+router.get('/loginServer/:email', authMiddleware, loginServer);
+
+// BORRAR!!!!
+router.get('/verifyEmail/:email', user.getByEmail);
+router.get('/', user.getAll);
```
Comment (BaltazarCortina) [inline routes/auth.js:20]: We should get rid of this

File: validators/candidates.js [hunk 7]
```diff
@@ -7,12 +7,12 @@ const isNotEmpty = (req, res, next) => {
...
-  }
-  if (!req.body.password) {
-    return res.status(400).json({ msg: 'Password is required' });
-  }
+  // if (!req.body.email) {
+  //   return res.status(400).json({ msg: 'Email is required' });
+  // }
+  // if (!req.body.password) {
+  //   return res.status(400).json({ msg: 'Password is required' });
+  // }
```
Comment (BaltazarCortina) [inline validators/candidates.js:15]: Should we delete this?

**Review by BaltazarCortina — APPROVED** [review]

**Review by juanandres7698 — APPROVED** [review]

**Pull request merged**